frontend/lib/context/KnowledgeProvider/knowledge-provider.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import { createContext, useState } from "react";
+
+import { Knowledge } from "@/lib/types/Knowledge";
+
+type KnowledgeContextType = {
+  allKnowledge: Knowledge[];
+  setAllKnowledge: React.Dispatch<React.SetStateAction<Knowledge[]>>;
+};
+
+export const KnowledgeContext = createContext<KnowledgeContextType | undefined>(
+  undefined
+);
+
+export const KnowledgeProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}): JSX.Element => {
+  const [allKnowledge, setAllKnowledge] = useState<Knowledge[]>([]);
```

Comment:
It appears that `allKnowledge` is not being used. A new state has been introduced in `useknowledgeTab`, which is conflicting with this

Right ?